test/position_calc_test/NLS_guess_distribution.py

```diff
@@ -0,0 +1,211 @@
+from components.position_calc.NLS_position_calc import NLS_position_calc, get_squared_error_sum
+from components.position_calc import position_calc_utils
+from simulator_main import sim_config as cfg
+from simulator_main import args
+from sim_utils.common_types import *
+import numpy as np
+import matplotlib.pyplot as plt
+from enum import Enum
+
+# determine which parts run
+vary_noise =        False
+vary_pinger =       False
+vary_guess =        False
+vary_hydrophones =  False
+
+class ChangingVariable(Enum):
+    '''
+    enum to decide the which variable is changing when plotting
+    '''
+    Noise = 0,
+    HydrophonePositions = 1,
+    PingerGuess = 2,
+    PingerPosition = 3
+
+# set global configuration
+cfg.speed_of_sound = 1500 #m/s
+cfg.hydrophone_positions = [
+    CylindricalPosition(0, 0, 0),
+    CylindricalPosition(3*DISTANCE_CONV["cm"], -np.pi/2, 0),
+    CylindricalPosition(3*DISTANCE_CONV["cm"], 0, 0),
+    CylindricalPosition(3*DISTANCE_CONV["cm"], np.pi/2, 0),
+    CylindricalPosition(3*DISTANCE_CONV["cm"], np.pi, 0),
+]
+cfg.pinger_position = CylindricalPosition(15, np.pi/5, 5)
+n_iter = 300
+
+def visualize_NLS_data(pinger_guess, noise_stdev, n_iters, change_var = ChangingVariable.Noise, 
+                        plot_error_hist=False):
+    '''
+    @brief visualizes the results of the NLS_position_calc component
+
+    @param pinger_guess     Initial guess for pinger position
+    @param noise_stdev      Standard deviation of gaussian noise injected to time differences
+                            (mean is 0)
+    @param n_iters          Number of noise simulations performed before plotting results
+    @param change_var       provides information about which variable is changing in simulation
+                            run to determine what information gets printed in plot title. Takes
+                            type ChangingVariable enum
+    @param plot_error_hist  flag to determine if to output a histogram of angular and distance error
+    '''
+    initial_guess = pinger_guess
+    opt_type = OptimizationType.nelder_mead
+    component = NLS_position_calc(optimization_type=opt_type, initial_guess=initial_guess)
+
+    true_tdoa = tuple(position_calc_utils.tdoa_function_3D(
+                np.array([cfg.pinger_position.r, cfg.pinger_position.phi]), hydrophone_position, True)
+                for hydrophone_position in cfg.hydrophone_positions[1:])
```

Comment:
I hate to be that dude in code reviews, but the indent here makes this kind of hard to read. 
IMO:
```
def _true_time_of_arrival(hydrophone_position):
    return position_calc_utils.tdoa_function_3D(
        np.array([cfg.pinger_position.r, cfg.pinger_position.phi]),
        hydrophone_position,
        True
    )

true_tdoa = tuple(
    _true_time_of_arrival(hydrophone_position)
    for hydrophone_position in cfg.hydrophone_positions[1:]
)
```

Reply:
At minimum I think
```
true_tdoa = tuple(
    position_calc_utils.tdoa_function_3D(np.array([cfg.pinger_position.r, cfg.pinger_position.phi]), 
                                         hydrophone_position, True)
    for hydrophone_position in cfg.hydrophone_positions[1:]
)
```